app/mailers/spree/base_mailer.rb

```diff
@@ -7,12 +7,9 @@ class BaseMailer < ActionMailer::Base
 
     helper TermsAndConditionsHelper
 
+    default from: -> { Spree::Config[:mails_from] }
```

Comment:
I needed a lambda here because the value changes during initialisation. Simplifying the mail config would solve this though.